chain/beacon/chain.go

```diff
@@ -217,15 +220,78 @@ func (c *chainStore) shouldSync(last *chain.Beacon, newB likeBeacon) bool {
 	return newB.GetRound() > last.GetRound()+1
 }
 
-// RunSync will sync up with other nodes and fill the store. If upTo is equal to
-// 0, then it will follow the chain indefinitely. If peers is nil, it uses the
-// peers of the current group.
-func (c *chainStore) RunSync(upTo uint64, peers []net.Peer) {
-	if peers == nil {
+// RunSync will sync up with other nodes and fill the store. If from is negative, it
+// will start from the latest stored beacon. If upTo is equal to 0, then it
+// will follow the chain indefinitely. If peers is nil, it uses the peers of
+// the current group.
+func (c *chainStore) RunSync(from, upTo uint64, peers []net.Peer) {
+	if len(peers) == 0 {
 		peers = toPeers(c.crypto.GetGroup().Nodes)
 	}
 
-	c.syncm.RequestSync(peers, upTo)
+	c.syncm.RequestSync(from, upTo, peers)
+}
+
+// ValidateChain will validate the chain in the chain store up to the given beacon.
+func (c *chainStore) ValidateChain(ctx context.Context, upTo uint64, cb func(r, u uint64)) ([]uint64, error) {
```

Comment:
It needs access to the verification logic, that's why it's here.

I'm not sure, but I think we could create a new boltdbstore since we're only reading the DB, even with the other main one still running, and wrap it with a SchemeStore and then do it as the sync manager level, maybe. WDYT?

Reply:
I changed it in [1cf9fc9](https://github.com/drand/drand/pull/1005/commits/1cf9fc9d5269e6d3fea6a60d93b4e542117d1343) after our call ^^'

Reply:
validation of individual beacons seems like a thing the chain store is responsible for. validation of ranges of the chain seems like something i would expect from the sync manager, since that's the only context i'd imagine that being done.

Reply:
Done in SyncManager now.